src/jit/morph.cpp

```diff
@@ -18519,6 +18521,280 @@ void Compiler::fgAddFieldSeqForZeroOffset(GenTree* op1, FieldSeqNode* fieldSeq)
+                // Keep track of the number of appearances of each promoted implicit
+                // byref (here during address-exposed analysis); fgMakeOutgoingStructArgCopy
+                // checks the ref counts for implicit byref params when deciding if it's legal
+                // to elide certain copies of them.
+                LclVarDsc* varDsc = &m_compiler->lvaTable[lclNum];
+                JITDUMP("Incrementing ref count from %d to %d for V%02d in %s\n", varDsc->lvRefCnt(RCS_EARLY),
+                        varDsc->lvRefCnt(RCS_EARLY) + 1, lclNum, __FUNCTION__);
+
+                varDsc->incLvRefCnt(1, RCS_EARLY);
+            }
+
+            if (node->OperIs(GT_LCL_FLD))
+            {
+                MorphLocalField(node, user);
+            }
+        }
+
+        PushValue(node);
+
+        return Compiler::WALK_CONTINUE;
+    }
+
+    fgWalkResult PostOrderVisit(GenTree** use, GenTree* user)
+    {
+        GenTree* node = *use;
+
+        switch (node->OperGet())
+        {
+            case GT_LCL_VAR:
+                assert(TopValue(0).Node() == node);
+                TopValue(0).Location(node->AsLclVar()->GetLclNum());
+                break;
+
+            case GT_LCL_FLD:
+                assert(TopValue(0).Node() == node);
+                TopValue(0).Location(node->AsLclFld()->GetLclNum(), node->AsLclFld()->gtLclOffs);
+                break;
+
+            case GT_ADDR:
+                assert(TopValue(1).Node() == node);
+                TopValue(1).Address(TopValue(0));
+                PopValue();
+                break;
+
+            case GT_FIELD:
+                if (node->AsField()->gtFldObj != nullptr)
+                {
+                    assert(TopValue(1).Node() == node);
+                    TopValue(1).Field(TopValue(0), node->AsField()->gtFldOffset);
+                    PopValue();
+                }
+                else
+                {
+                    assert(TopValue(0).Node() == node);
+                }
+                break;
+
+            case GT_OBJ:
+            case GT_BLK:
+            case GT_IND:
+                assert(TopValue(1).Node() == node);
+                TopValue(1).Indir(TopValue(0));
+                PopValue();
+                break;
+
+            case GT_CALL:
+                for (; TopValue(0).Node() != node; PopValue())
+                {
+                    if (TopValue(0).IsAddress())
+                    {
+                        bool fieldOnly = (TopValue(0).Node() == node->AsCall()->gtCallObjp);
+                        MarkAddressExposed(TopValue(0), fieldOnly);
+                    }
+                }
+                break;
+
+            default:
+                for (; TopValue(0).Node() != node; PopValue())
+                {
+                    if (TopValue(0).IsAddress())
+                    {
+                        MarkAddressExposed(TopValue(0));
+                    }
+                }
+                break;
+        }
```

Comment:
Do we need a case for  `GT_DYN_BLK`?

Reply:
I don't think so, it does not have a known size so the best we can do is to treat it as any other unknown node and escape the address.

Reply:
Hmm, maybe I'll add it for the sake of consistency.